app/javascript/components/subnet-form/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './subnet-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import { API } from '../../http_api';
+
+const SubnetForm = ({ recordId }) => {
+  const [{ initialValues, isLoading, fields }, setState] = useState({ isLoading: !!recordId, fields: [] });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const loadSchema = (appendState = {}) => ({ data: { form_schema: { fields } } }) => {
+    if (!!recordId && appendState.initialValues.type === 'ManageIQ::Providers::Openstack::NetworkManager::CloudSubnet') {
+      Object.assign(fields[0], {isDisabled: true});
+      Object.assign(fields[1], {isDisabled: true});
+      Object.assign(fields[4], {isDisabled: true});
+    }
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/cloud_subnets/${recordId}`).then((initialValues) => {
+        if(typeof initialValues.cloud_network_id ==="string") {
+          initialValues.cloud_network_id=Number(initialValues.cloud_network_id);
+        }
+        if(typeof initialValues.cloud_tenant_id ==="string") {
+          initialValues.cloud_tenant_id=Number(initialValues.cloud_tenant_id);
+        }
+        API.options(`/api/cloud_subnets?ems_id=${initialValues.ems_id}`).then(loadSchema({ initialValues, isLoading: false }));
+      });
+    }
+  }, [recordId]);
+
+  const onSubmit = (values) => {
+    API.get(`/api/providers/${values.ems_id}`).then(({ type }) => {
+      if (type === 'ManageIQ::Providers::Openstack::NetworkManager') {
+        if (values.ip_version === undefined) {
+          values.ip_version = '4';
+        }
+        if (values.dhcp_enabled === undefined) {
+          values.dhcp_enabled = false;
+        }
+        values.enable_dhcp = values.dhcp_enabled;
+        delete values.dhcp_enabled;
+        delete Object.assign(values, values.extra_attributes).extra_attributes;
+
+        API.get(`/api/cloud_networks/${values.cloud_network_id}`).then( ({ ems_ref }) => {
+          values.network_id = ems_ref;
```

Comment:
the `onSubmit` function is wrapped in two API calls which are needed to retrieve the `ems_ref` related to the network chosen by the user.

Reply:
would it be possible to instead grab `ems_ref` here? https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb#L123-L129

Reply:
mat be check with @agrare

Reply:
If you want to pass in the cloud_network_id then translate that to `cloud_network = CloudNetwork.find_by(:id => cloud_network_id); ems_ref = cloud_network.ems_ref` in the provider yes you can do that